extensions/keycloak-authorization/runtime/src/main/java/io/quarkus/keycloak/pep/runtime/KeycloakPolicyEnforcerRecorder.java

```diff
@@ -26,14 +33,26 @@
 
 @Recorder
 public class KeycloakPolicyEnforcerRecorder {
-    final HttpConfiguration httpConfiguration;
 
-    public KeycloakPolicyEnforcerRecorder(HttpConfiguration httpConfiguration) {
-        this.httpConfiguration = httpConfiguration;
+    public BooleanSupplier createBodyHandlerRequiredEvaluator(KeycloakPolicyEnforcerConfig config) {
+        return new BooleanSupplier() {
```

Comment:
@michalvavrik Can you clarify please how it fixes the original problem, related to handling the body, which was what tying the config to the build time, CC @pedroigor

Reply:
router is created as one of last things (even though build step order is not deterministic I guess) and even if it were not, you can see the build step order allows us to access runtime config during creation of routes. Which means we can just postpone that check whether we need it or not until routes are actually added.

the point is, it is determined during the runtime anyway, so instead of signalling it by some build time, we signal it by the supplier. it is backwards compatible solution and I don't think anyone could be affected by this.